express-post-json/index.js

```diff
@@ -0,0 +1,27 @@
+const express = require('express');
+const app = express();
+
+let nextId = 1;
+const grades = {};
+
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
+  const gArray = [];
+  for (const id in grades) {
+    gArray.push(grades[id]);
+  }
+  res.json(gArray);
+});
+
+app.post('/api/grades', (req, res) => {
+  const data = req.body;
+  data.id = nextId;
+  res.status(201).send(grades[data.id] = data);
+  nextId++;
```

Comment:
Although this works it is best to have the send line as the final line in the route handler, you don't want the server still doing stuff after a response has been sent. Also remember to use `.json` over `.send` and finally doing assignment and passing to a function  works but is generally not the best practice. So these two lines should be refactored to something like:

```JAVASCRIPT
grades[data.id] = data;
nextId++;
res.status(201).json(data);
```